### DISCUSSION TITLE
Duvida em criação de componente

### DISCUSSION THREAD

**Discussion by felipevc13** [frontendbr/forum#1955] [Perguntas e Respostas] [answered]
Fala pessoal blz?
Sou novo na área e estou fazendo alguns estudos criando um app de alimentação para o meu cachorro. 

Eu tenho um componente que vai ter 3 estados:

Estado inicial: O componente quando não tem nenhum item vindo do banco de dados ele vai ter uma mensagem.

Estado de edição: O componente vai adicionar, editar e excluir itens.

Estado com dados: Estado onde os itens aparecem em uma tabela.

A pergunta é, como vocês fariam a separação desse componente e quais argumentos passariam para ele? 

Estou usando o Gatsby + Strapi e Graphql para pegar os dados.

Qualquer dica é bem vinda.

Muito Obrigado.

<img width="393" alt="Captura de Tela 2020-12-22 às 12 52 13" src="https://user-images.githubusercontent.com/25945473/102907605-129b2b00-4455-11eb-9988-cb3d9fa703ac.png">

**Comment by victorferreira** [comment, marked as answer]
Como no seu caso você está sempre editando a lista inteira, você pode fazer algo como: 

```javascript
function Index() {
  const [list, setList] = useState([]);
  const [isEditing, setIsEditing] = useState(false);

  if (isEditing) {
    return <EditForm list={list} />;
  }

  if (!list.length) {
    return <EmptyList onUpdate={setList} openEditForm={setIsEditing} />;
  }

  return (
    <PopulatedList list={list} onUpdate={setList} openEditForm={setIsEditing} />
  );
}
```

**Reply by felipevc13** [reply]
muito obrigado pela resposta, me ajudou muito.
